packages/code-du-travail-data/dataset/fiches_ministere_travail/referenceExtractor.js

```diff
@@ -0,0 +1,244 @@
+/*
+Extracting references is done in several steps :
+1) classifyTokens : we identify valid article references (start with l/r/d then token of shape like 1234-12) => split text into sequence of tokens and give a label to each token
+2) identifyCode : we search for the associated code after the ref tokens (valid options are : code du travail / code de la scurite sociale)
+3) we group those to constitute structured reference of shape : 
+  {
+    "article": "L. 2313-8",
+    "code": Object {
+      "id": "LEGITEXT000006072050",
+      "name": "code du travail",
+    },
+  }
+*/
+
+const treebank = require("talisman/tokenizers/words/treebank");
+
+const NEGATIVE = "O";
+const ARTICLE = "B-ART";
+const CODE_PREFIX = "B-COD";
+// code du travail
+const CODE_TRA = CODE_PREFIX + "_TRA";
+// code sécurité sociale
+const CODE_SS = CODE_PREFIX + "_SS";
+
+const CODE_TRAVAIL = {
+  name: "code du travail",
+  id: "LEGITEXT000006072050"
+};
+
+const codesFullNames = {
+  [CODE_SS]: {
+    name: "code de la sécurité sociale",
+    id: "LEGITEXT000006073189"
+  },
+  [CODE_TRA]: CODE_TRAVAIL
+};
+
+// maximum distance between code tokens and corresponding article ref
+const range = 20;
+
+const articleRegEx = new RegExp("^(\\d{1,4}(-\\d+){0,3})\\b"); //          nums        123 123-45 123-45-6 123-45-6-7
+function articleMatcher(token) {
+  return token.match(articleRegEx);
+}
+
+const validPrefix = ["l", "r", "d"];
+
+// returns :
+// 0 if not matching
+// 1 if matching prefix only (L.)
+// 2 if matching prefix and valid ref (L123.12)
+function prefixMatcher(token) {
+  const lowToken = token.toLowerCase();
+
+  // if starts with possible prefix
+  const matchingPrefix =
+    validPrefix.filter(p => lowToken.startsWith(p)).length > 0;
+
+  if (matchingPrefix) {
+    const residual = lowToken.slice(1);
+
+    // case only L
+    if (!residual.length) {
+      return 1;
+    } else {
+      // case L.
+      if (residual == ".") {
+        return 1;
+      } else {
+        // case L.123-12
+        if (residual.slice(0, 1) == "." && articleMatcher(residual.slice(1))) {
+          return 2;
+        }
+        // case L.123-12
+        else if (articleMatcher(residual.slice(1))) {
+          return 2;
+        }
+      }
+    }
+  }
+  // no match
+  return 0;
+}
+
+function infixMatcher(token) {
+  // this is quite subtle...
```

Comment:
don't see the différence between the two letter, (it's subtle) can you explain why ?